examples/lists.rs

```diff
@@ -0,0 +1,49 @@
+#[macro_use]
+extern crate redis_module;
+
+use redis_module::raw::KeyType;
+use redis_module::{Context, RedisError, RedisResult, RedisValue};
+
+// LPOPRPUSH source destination
+// Pops and returns the first element (head) of the list stored at 'source'
+// and pushes the element to the last position (tail) of the list stored at
+// 'destination'.
+fn lpoprpush(ctx: &Context, args: Vec<String>) -> RedisResult {
+    if args.len() != 3 {
+        return Err(RedisError::WrongArity);
+    }
+
+    let src_key = ctx.open_key_writable(&args[1]);
+    let dst_key = ctx.open_key_writable(&args[2]);
+
+    let src_type = src_key.key_type();
+    let dst_type = dst_key.key_type();
+
+    if (src_type != KeyType::Empty && src_type != KeyType::List)
+        || (dst_type != KeyType::Empty && dst_type != KeyType::List)
+    {
+        return Err(RedisError::WrongType);
+    }
+
+    match src_key.list_pop_head() {
+        None => {
+            return Ok(RedisValue::Null);
+        }
+        Some(value) => {
+            let ret_cpy = value.clone();
+            dst_key.list_push_tail(value);
+            return Ok(RedisValue::BulkString(ret_cpy.into()));
+        }
```

Comment:
This can be simplified a bit:
```suggestion
        None => Ok(RedisValue::Null),
        Some(value) => {
            let ret_cpy = value.clone();
            dst_key.list_push_tail(value);
            Ok(RedisValue::BulkString(ret_cpy.into()))
        }
```